### PR TITLE
fix: change documentation for using nix-shell with remote url

### LINKED ISSUE #139: error: nix-shell requires a single derivation
I am trying to build ghc with nix.

My ghc repository is up to date.

I am running this:

```
$ nix-shell https://github.com/alpmestan/ghc.nix/archive/master.tar.gz
error: nix-shell requires a single derivation
Try 'nix-shell --help' for more information.
```
on ghc.nix commit: 133516af8426d775fa0dc75c787edd56299ee6cf

Its a while since I used it but then it worked.
Am I missing something?

### PR DESCRIPTION
**Pull request by MangoIV**
fixes #139 (refer to the discussion there for information of why this change is necessary)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,7 +11,7 @@ an excellent place to start.
 To enter an environment without cloning this repository you can run:
 
 ```
-nix-shell https://github.com/alpmestan/ghc.nix/archive/master.tar.gz
+nix-shell https://github.com/alpmestan/ghc.nix/archive/master.tar.gz --attr devShells.<your-system>.default
```
Comment (supersven) [inline README.md:14]: I think an explanation what `<your-system>` could be, would be good. Likely with an example.

**Comment by MangoIV** [comment]
@alpmestan are you happy with just supplying documentation of how to invoke the devShell with the tarball URL? This has two shortcomings
- we always have to supply the `--arg` argument to `nix-shell` when trying to build directly from the url
- we cannot supply arguments as the `default.nix` does not expect arguments (see my other PR) 

Alternatively we can go back to the "old" design which is just providing the same interface as in the `shell.nix` in the `default.nix`, too. This would, however, make it odd to use for people who're used to `flake-compat` and we would not have access to the flake attributes from the legacy nix code. To ship around this, we could perhaps also add another file e.g. `legacy-flake.nix` that offers the interface that the `default.nix` offers now.

File: README.md (hunk 1 above)
Comment (MangoIV) [reply README.md:14]: Thank you, I have fixed this.

**Comment by MangoIV** [comment]
oh, I'm sorry for pinging the wrong person, I guess reading the name every time made my brain do things... Sorry, @supersven, this comment was directed to you...

**Review by supersven — APPROVED** [review]
@MangoIV Thanks!

LGTM :+1:

**Pull request merged**